### ISSUE TITLE
CI build is taking ages

### ISSUE THREAD

**Issue by gashcrumb** [syndesisio/syndesis#3563]
Look at this -> https://circleci.com/workflow-run/f4f09b38-8dde-40c3-851d-3112109b6c14

I don't recall builds taking that long but an hour and 46 minutes seems a bit long => https://circleci.com/workflow-run/1857a9fe-8ec7-4fd3-8aef-c29cc68848b2

Why does the CI build take so long?

**Comment by gashcrumb** [comment]
@paoloantinori wonder, who looks after the builds these days?

**Comment by rhuss** [comment]
@gashcrumb might be because we switched from the free plan with 4 parallel containers to a paid plan with 2 containers. If there are several PRs running in parallel it might be that one of the test is kicked of later. 

You should check the workflow and the duration of the individual build steps. If these sound reasonable but the overall time takes too long, then the gap between steps is because there is no free container for this step and has to wait until another step is finished (which is possible one from another PR).

> @paoloantinori wonder, who looks after the builds these days?

I still hope this is a team effort without a dedicate 'CI guard'. Actually I added some IRC integration which doesn't seem to really work, but it was intended to post a message when a build fails. will check that later.

**Comment by rhuss** [comment]
Looking a bit closer to the referenced build, it indeed looks like the build steps themselves took that long. Looking into the logs it looks like that every Maven dependenncy is downloaded, which looks like that either our caching mechanism for .m2 doesnt work or CircleCI has either invalidated our cache or has another issues with the caching mechanism.

I suggest to wait some build and if it still downloads the wordl for every build recheck our CI scripts.

**Comment by gashcrumb** [comment]
I wasn't sure.  20 min for the UI build for example seems a bit lengthy.  Connector/integration seem to take quite awhile too, and they need to be built sequentially.  Is the maven repo getting cached for these builds I wonder?

**Comment by rhuss** [comment]
Yes, we always cache .m2 across steps (thats also why we have so many sequential steps as that cache can't be really shared)

**Comment by rhuss** [comment]
Just checked again: Each step has its own cache which gets invalidated when the pom.xml changes (via a checksum). See e.g. https://github.com/syndesisio/syndesis/blob/543cb8eb6e6812daf4baf090430fa8b1e4d68f9e/.circleci/config.yml#L309-L312 for saving the cache and https://github.com/syndesisio/syndesis/blob/543cb8eb6e6812daf4baf090430fa8b1e4d68f9e/.circleci/config.yml#L284-L285 for restoring.

So any time the pom.xml changes the build is supposed to take much longer.

**Comment by heiko-braun** [comment]
@gashcrumb Has this been resolved?
